### PR TITLE
Switch from string matching to registry key value for activation status

### PR DESCRIPTION
**Pull request by jjerger**
More deterministic; other method relies on system display language being English.

### CODE REVIEW

**Comment by jjerger** [comment]
Holding while I test in Server 2016, 2019, and 2022.

Confirmed already working in 2022, but I noticed the key is absent in 2016 until registration is attempted. Making sure this won't break the script.

File: sysprep/activate_instance.ps1 [hunk 1]
```diff
@@ -264,18 +264,16 @@ function Verify-ActivationStatus {
...
+    $activation_status = (Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\SoftwareProtectionPlatform\Activation" -Name ProductActivationResult).ProductActivationResult
   }
   catch {
     return $active
   }
 
-  $status = $slmgr_status | Select-String -Pattern '^License Status:'
-  # The initial space is to ensure "Unlicensed" does not match.
-  if ($status -match ' Licensed') {
+  # Anything other than 0x0 is a failure.
```
Comment (bkatyl) [inline sysprep/activate_instance.ps1:275]: I'd suggest keeping both method. 

If the regkey matches don't do the string match.
if the rekey doesn't match, do the string match attempt as a fall through.

Comment (jjerger) [reply sysprep/activate_instance.ps1:275]: Moved this to a new PR cause the old branch was so old it was better to just rebase a new one
https://github.com/GoogleCloudPlatform/compute-image-windows/pull/252

**Pull request closed without merging**